packages/markdown/remark/src/remark-collect-images.ts

```diff
@@ -1,14 +1,24 @@
-import type { Image } from 'mdast';
+import type { Image, ImageReference } from 'mdast';
+import { definitions } from 'mdast-util-definitions';
 import { visit } from 'unist-util-visit';
 import type { MarkdownVFile } from './types';
 
 export function remarkCollectImages() {
 	return function (tree: any, vfile: MarkdownVFile) {
 		if (typeof vfile?.path !== 'string') return;
 
+		const definition = definitions(tree);
 		const imagePaths = new Set<string>();
-		visit(tree, 'image', (node: Image) => {
-			if (shouldOptimizeImage(node.url)) imagePaths.add(node.url);
+		visit(tree, ['image', 'imageReference'], (node: Image | ImageReference) => {
+			if (node.type === 'image') {
+				if (shouldOptimizeImage(node.url)) imagePaths.add(node.url);
+			}
+			if (node.type === 'imageReference') {
+				const imageDefinition = definition(node.identifier);
```

Comment:
Instead of pre-loading the `definitions(tree)`, we could also look them up lazily here. This would lower the performance impact when not using the additional `imageReference` syntax (the majority I guess), but slow down the implementation in this PR when that syntax is used.

Either way, this type of AST traversal is fast. We could traverse only direct `tree.children` (no further descendants) to minimize overhead.